e2e/autoscript-suites/src/suites/authn-oauth.lc.test.ts

```diff
@@ -31,4 +31,51 @@ test.describe('Test OAuth login flow', () => {
     );
     expect(revokeRequests.length).toBe(2);
   });
+
+  test(`should proactively refresh tokens if they expire within default threshold`, async ({
+    page,
+    browserName,
+  }) => {
+    const { messageArray } = await setupAndGo(page, browserName, 'authn-oauth/', {
+      realmPath: 'tokens-expiring-soon',
+    });
+    expect(messageArray.includes('OAuth tokens expiring soon; proactively refreshed')).toBe(true);
+  });
+
+  test(`should proactively refresh tokens if they expire within specified threshold`, async ({
+    page,
+    browserName,
+  }) => {
+    const { messageArray } = await setupAndGo(page, browserName, 'authn-oauth/', {
+      realmPath: 'tokens-expiring-soon',
+      oauthThreshold: '25000',
+    });
+    expect(messageArray.includes('OAuth tokens expiring soon; proactively refreshed')).toBe(true);
+  });
+
+  test(`should not proactively refresh tokens if they expire outside default threshold`, async ({
+    page,
+    browserName,
+  }) => {
+    const { messageArray } = await setupAndGo(page, browserName, 'authn-oauth/', {});
+    expect(messageArray.includes('OAuth tokens not expiring soon; not refreshed')).toBe(true);
+  });
+
+  test(`should not proactively refresh tokens if they expire outside specified threshold`, async ({
+    page,
+    browserName,
+  }) => {
+    const { messageArray } = await setupAndGo(page, browserName, 'authn-oauth/', {
+      realmPath: 'tokens-expiring-soon',
+      oauthThreshold: '10000',
+    });
+    expect(messageArray.includes('OAuth tokens not expiring soon; not refreshed')).toBe(true);
+  });
```

Comment:
There are 5 newly introduced tests here, which means 5 independent browser launches. Can we reuse the same browser lunch for one or more of these? We should be able to, at least, move this one into the original OAuth test, yeah? Having independent browser launches if not bad, per se, but it does come with a performance liability at scale.

Reply:
Ok, I see your point about multiple browser launches. I think we can combine a few of these by having multiple **setupAndgo** calls within each **test** block. I'll have a go at rearranging these accordingly.

Reply:
I've grouped the OAuth tests into three browser launches - one for each type of test outcome as it felt like it made sense. I could however reduce further and just have one launch for the whole test script. What do you think?

Reply:
It's a judgement call, and I'll leave it up to you to figure out what's the right number. I don't want to us to be overly concerned about separate browser launches, but I just want us to be mindful.

Reply:
Understood. Thanks